lib/meta.nix

```diff
@@ -99,4 +99,27 @@ rec {
   availableOn = platform: pkg:
     lib.any (platformMatch platform) pkg.meta.platforms &&
     lib.all (elem: !platformMatch platform elem) (pkg.meta.badPlatforms or []);
+
+  /* Get the corresponding attribute in lib.licenses
+     from the SPDX ID.
+     For SPDX IDs, see
+     https://spdx.org/licenses
+
+     Type:
+       getLicenseFromSpdxId :: str -> AttrSet
+
+     Example:
+       lib.getLicenseFromSpdxId "MIT" == lib.licenses.mit
+       => true
+       lib.getLicenseFromSpdxId "mIt" == lib.licenses.mit
+       => true
+       lib.getLicenseFromSpdxId "MY LICENSE"
+       => { shortName = "MY LICENSE"; }
```

Comment:
Hmm, shouldn't this return an error or an indication that the string doesn't represent a known SPDX license?

Reply:
Updated to make it warn for no matching SPDX ID.
Considering that the main use case of this function is to parse the license information from `package.json` or similar configuration files, there need to be a fallback in case the string doesn't match a known SPDX ID.